camel/models/azure_openai_model.py

```diff
@@ -0,0 +1,134 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+from typing import Any, Dict, List, Optional, Union
+
+from openai import AzureOpenAI, Stream
+
+from camel.configs import OPENAI_API_PARAMS
+from camel.messages import OpenAIMessage
+from camel.models.base_model import BaseModelBackend
+from camel.types import ChatCompletion, ChatCompletionChunk, ModelType
+from camel.utils import BaseTokenCounter, OpenAITokenCounter, api_keys_required
+
+
+class AzureOpenAIModel(BaseModelBackend):
+    r"""Azure OpenAI API in a unified BaseModelBackend interface.
+    Doc: https://learn.microsoft.com/en-us/azure/ai-services/openai/
+    """
+
+    def __init__(
+        self,
+        model_type: ModelType,
+        model_config_dict: Dict[str, Any],
+        api_key: Optional[str] = None,
+        url: Optional[str] = None,
+    ) -> None:
+        r"""Constructor for OpenAI backend.
+
+        Args:
+            model_type (ModelType): Model for which a backend is created,
+                one of GPT_* series.
+            model_config_dict (Dict[str, Any]): A dictionary that will
+                be fed into openai.ChatCompletion.create().
+            api_key (Optional[str]): The API key for authenticating with the
+                OpenAI service. (default: :obj:`None`)
+            url (Optional[str]): The url to the OpenAI service. (default:
+                :obj:`None`)
+        """
+        super().__init__(model_type, model_config_dict, api_key, url)
+        self._url = url or os.environ.get("AZURE_OPENAI_ENDPOINT")
+        self._api_key = api_key or os.environ.get("AZURE_OPENAI_API_KEY")
+        self.api_version = os.environ.get("AZURE_API_VERSION")
+        self.azure_deployment = os.environ.get("AZURE_DEPLOYMENT")
+
+        if self._url is None:
+            raise ValueError("url or `AZURE_OPENAI_ENDPOINT` is not provided.")
+        if self._api_key is None:
+            raise ValueError(
+                "api_key or `AZURE_OPENAI_API_KEY` is not provided."
+            )
+        if self.api_version is None:
+            raise ValueError("`AZURE_API_VERSION` is not provided.")
+        if self.azure_deployment is None:
+            raise ValueError("`AZURE_DEPLOYMENT` is not provided.")
+        self.model = str(self.azure_deployment)
+
+        self._client = AzureOpenAI(
+            azure_endpoint=str(self._url),
+            api_version=self.api_version,
+            api_key=self._api_key,
+            timeout=60,
+            max_retries=3,
+        )
```

Comment:
maybe need add deploy model name?

Reply:
the deploy model name is set to AZURE_DEPLOYMENT.